### ISSUE TITLE
Add cache directory (or tldr checkout directory) to config

### ISSUE THREAD

**Issue by slippycheeze** [tealdeer-rs/tealdeer#156]
G'day.  I use a couple different tldr clients, and so rather than maintaining multiple caches I just keep my own git checkout around and updated regularly.

I can get tealdeer to use that by way of `export TEALDEER_CACHE_DIR="${HOME}/var/cache"` and naming the checkout `tldr-master` under it.  Which is kind of ugly, but whatever.

It'd be great if I could just add `cache = ~/var/cache` (or, better, `tldr = ~/var/cache/tldr`) to the config.toml and never have to worry about the environment variable again.  (in the second example, that'd be the path to the directory with the content, so that you didn't need append anything to the directory to get the page root.)

**Comment by niklasmohrin** [comment]
We could also consider just unpacking the tldr repo directly into the `TEALDEER_CACHE_DIR`. This is only an option, if we don't need to put anything else in the cache, but I wouldn't know what (caching of rendered pages for performance incoming? 😆).

**Comment by hgaiser** [comment]
Hopefully https://github.com/dbrgn/tealdeer/issues/156 will fix this issue :).

**Comment by dbrgn** [comment]
Right now, the cache directory is expected to be a cache under the control of tealdeer. This is shown for example in the PR #206, where we are adding a migration that renames the checkout directory, and - if both exist - deletes the old directory.

If we add the option to externally manage the tldr repo checkout, then enabling this feature should disable all cache management functionality in tealdeer to avoid conflicts. However, that shouldn't be the same as configuring the cache directory. The directory layout inside the cache directory is _not_ an API, and I do not want to guarantee a specific structure.

I do agree that `TEALDEER_CACHE_DIR` is a bit of an outlier. Initially we just had this env variable to configure tealdeer, and no config file. Then we added the config file. Now that this config file exists, there's no reason why the cache directory should be configured through an env var, and not through the config.

TLDR:

- Making the cache directory configurable through the config file makes sense. The env var should then be deprecated (and a deprecation warning should be shown if it's still set).
- We do not guarantee any specific layout inside this cache directory, and tealdeer may modify/rename/delete files inside that directory at any time. Pointing the cache directory to a shared dir is risky.
- If people would like to manage the tldr pages themselves, this should be a separate option and it should disable all cache management functionality. However, I suspect that this would add significant complexity to the sourcecode, and I'd be reluctant to merge such a PR.

**Comment by niklasmohrin** [comment]
This was implemented in #276 :)

**Comment by niklasmohrin** [comment]
Although, regarding the initial use case, we still do not guarantee anything about the contents of the cache. In fact, we are planning to switch to the language specific page archives soon (#335), which will result in a slightly altered layout where the `pages` directory will be renamed to `pages.en`.
